gate: a bare || collapses patch-coverage exit 3 (REFUSED) into "below threshold", pointing at the wrong remedy

<!-- Agent hints: set all three before submitting -->
[mode: direct]
[model: sonnet]
[effort: medium]

## Description

`scripts/pre-push-gate.sh:101` collapses every non-zero exit from the patch-coverage estimator into one message:

```sh
COVER_OUT="$COVER_OUT" \
  bash "$HELPER" || fail "patch coverage below codecov.yml threshold (note: this gate is a conservative lower bound; Codecov typically reads a few points higher)"
```

A bare `||` cannot distinguish the estimator's exit codes, and it defines four with **different meanings and opposite remedies**:

| Exit | Meaning | Correct remedy |
|---|---|---|
| 0 | Passed | -- |
| 1 | Below threshold | **Write tests** |
| 2 | Setup/config error | Fix the config or profile |
| 3 | **REFUSED** -- validity precondition failed | **Commit your changes** |

Only exit 1 means what the gate prints. Exits 2 and 3 both report "patch coverage below codecov.yml threshold", which is false, and which sends the reader to do work that cannot help.

### Observed

Hit on PR #766 while committing review fixes. The gate printed:

```
FAIL: patch coverage below codecov.yml threshold (note: this gate is a conservative
lower bound; Codecov typically reads a few points higher)
make: *** [gate] Error 1
```

The estimator had actually printed, immediately above, in the same output:

```
patch-coverage: REFUSED (exit 3) -- uncommitted Go changes are present.
  A patch-coverage figure is valid only at a committed HEAD with a clean tree
  ...
  Commit first, then re-run.
```

Coverage was never measured -- it was 100% (22/22) on the very next run, once the files were committed, with no test changes at all. Writing tests in response to the printed message would have been pure waste.

### Why the mislabel is worse than a missing message

The estimator's own diagnostic is correct and detailed. The gate then **contradicts it** in the last line, which is the line a reader trusts: it is the FAIL, it is what `make gate` exits on, and in CI-style scrollback it is often all that survives. A reader who trusts the summary over the body reaches for the wrong tool.

This is also the failure mode most likely to hit during a review-fix round, when the tree is dirty by definition -- exactly when the gate is being run under time pressure.

## Proposed Solution

Branch on the exit code and print the estimator's own remedy:

```sh
if [ -x "$HELPER" ]; then
  set +e
  COVER_OUT="$COVER_OUT" bash "$HELPER"
  pc_rc=$?
  set -e
  case "$pc_rc" in
    0) ;;
    1) fail "patch coverage below codecov.yml threshold (conservative lower bound; Codecov typically reads a few points higher)" ;;
    3) fail "patch coverage REFUSED -- see the estimator's message above; it was NOT measured. Usually: commit your changes, then re-run." ;;
    *) fail "patch coverage estimator error (exit $pc_rc) -- see its message above; coverage was NOT measured" ;;
  esac
fi
```

Note the `set +e` / `set -e` dance is needed because the gate runs under `set -e`; capturing `$?` after a bare call would otherwise abort before the `case`.

Worth checking whether `coverage-floor.sh` (line ~118) and any other `|| fail`-wrapped helper in the gate have the same shape. The floor step already distinguishes its exit 1 from other codes, so it is the pattern to copy rather than a second instance of the bug.

## Acceptance Criteria

- [ ] Exit 3 reports REFUSED and states coverage was not measured, not a threshold failure
- [ ] Exit 2 (and any other unexpected code) reports an estimator error rather than a threshold failure
- [ ] Exit 1 keeps its current message
- [ ] Verified by running the gate with a deliberately dirty tree and reading the final FAIL line
- [ ] Every other `|| fail`-wrapped helper in `pre-push-gate.sh` audited for the same collapse

## Additional Context

- Estimator exit codes are documented at `~/.claude/scripts/patch-coverage.sh:120` (claude-kit, not this repo), so this repo consumes a contract it does not own -- an argument for handling unknown codes explicitly rather than assuming the set stays at four.
- The gate is otherwise correct to treat a missing estimator as SKIP; that part needs no change.

--- a/internal/queue/queue.go
+++ b/internal/queue/queue.go
@@ -1309,13 +1309,37 @@ func (q *DBQueue) RecheckDeferred(ctx context.Context, libraryID *int64) (int64,
 }
 
 // CountRecheckDeferred returns the number of 'deferred' rows that
-// RecheckDeferred would revive, without writing. Intended for dry-run output.
+// RecheckDeferred would actually CHANGE, without writing. Intended for dry-run
+// output.
+//
+// THE PREDICATE MUST MATCH THE APPLY'S EFFECT, not merely its WHERE clause
+// (#774). RecheckDeferred writes next_attempt_at and deliberately leaves status
+// alone, so counting on status alone described a set that a successful revive
+// could not shrink: re-running the dry run reported the identical number, which
+// is misleading at exactly the moment an operator re-runs it to confirm the apply
+// worked. Measured on a live instance: 18,651 revived, then 18,651 "would
+// revive". Adding the next_attempt_at bound makes the count a genuine preview --
+// it counts rows still waiting out a retry delay, which is what reviving them
+// changes.
+//
+// Note this deliberately does NOT match RecheckDeferred's own WHERE clause, which
+// stays broader: re-stamping an already-eligible row is harmless and idempotent,
+// so the apply has no reason to exclude it, while the COUNT must report only rows
+// whose state the apply would move. The two agree on the population that matters
+// and differ only on rows for which the write is a no-op.
+//
+// The 'now' here is q.now(), the same clock the apply stamps with, so a frozen
+// test clock keeps both sides consistent.
+//
+// The retired twin needs no equivalent: RecheckRetired's count and apply share
+// status='done' AND last_error=<sentinel>, and the apply MOVES status, so that
+// count already shrinks on a successful revive.
 func (q *DBQueue) CountRecheckDeferred(ctx context.Context, libraryID *int64) (int64, error) {
 	libClause, libArgs := recheckLibraryClause(libraryID)
-	args := append([]any(nil), libArgs...)
+	args := append([]any{formatTime(q.now())}, libArgs...)
 	var count int64
 	if err := q.db.QueryRowContext(ctx,
-		`SELECT COUNT(*) FROM work_queue WHERE status = 'deferred'`+libClause, //nolint:gosec // G202: libClause is a hardcoded constant from recheckLibraryClause, never user input
+		`SELECT COUNT(*) FROM work_queue WHERE status = 'deferred' AND next_attempt_at > ?`+libClause, //nolint:gosec // G202: libClause is a hardcoded constant from recheckLibraryClause, never user input
 		args...,
 	).Scan(&count); err != nil {
 		return 0, fmt.Errorf("queue: count recheck deferred: %w", err)

--- a/internal/queue/queue_test.go
+++ b/internal/queue/queue_test.go
@@ -2573,6 +2573,24 @@ func TestDBQueue_CountRecheckDeferred(t *testing.T) {
 	if n != count {
 		t.Fatalf("RecheckDeferred = %d; CountRecheckDeferred = %d; must agree", n, count)
 	}
+
+	// THE POINT OF #774: the count must be a PREVIEW of the apply, which means it
+	// has to MOVE once the apply succeeds. Asserting agreement BEFORE the apply
+	// (above) cannot catch this -- both sides read the same rows at that moment.
+	//
+	// Before the fix the count read only `status`, while the apply wrote
+	// `next_attempt_at` and deliberately left `status` alone, so a successful
+	// revive changed nothing the count could see. Re-running the dry run reported
+	// the identical number, which is misleading at exactly the moment an operator
+	// re-runs it to confirm the apply worked. Measured on a live instance: 18,651
+	// revived, then 18,651 "would revive".
+	after, err := q.CountRecheckDeferred(ctx, nil)
+	if err != nil {
+		t.Fatalf("CountRecheckDeferred after apply: %v", err)
+	}
+	if after != 0 {
+		t.Errorf("count after a successful revive = %d; want 0 (the dry run must reflect the apply)", after)
+	}
 }
 
 // TestDBQueue_RecheckRetired verifies that RecheckRetired revives

--- a/scripts/pre-push-gate.sh
+++ b/scripts/pre-push-gate.sh
@@ -95,10 +95,52 @@ echo "==> patch coverage (Codecov parity, conservative lower bound)"
 # result; when absent it is SKIPPED, not failed. This gate is a dev-only
 # convenience -- CI enforces patch coverage via Codecov directly
 # (.github/workflows/ci.yml), so nothing is lost when the estimator is missing.
-HELPER="$HOME/.claude/scripts/patch-coverage.sh"
+# Overridable so the exit-code branches below can be exercised against a stub,
+# following the TAILWIND_BIN convention above. Unset -> the real estimator.
+HELPER="${PATCH_COVERAGE_HELPER:-$HOME/.claude/scripts/patch-coverage.sh}"
 if [ -x "$HELPER" ]; then
-  COVER_OUT="$COVER_OUT" \
-    bash "$HELPER" || fail "patch coverage below codecov.yml threshold (note: this gate is a conservative lower bound; Codecov typically reads a few points higher)"
+  # BRANCH ON THE EXIT CODE, never a bare `||` (#768). The estimator defines four
+  # codes whose remedies DIFFER, and two of them are OPPOSITE: exit 1 means write
+  # tests, exit 3 means commit what you already wrote. A bare `||` collapsed all of
+  # them into "below threshold", which is false for 2 and 3 and sends the reader to
+  # do work that cannot help. Observed three times in one session, costing a cycle
+  # each: the message is confident, plausible, and wrong.
+  #
+  # `rc=0; ... || rc=$?` rather than calling it bare: `set -e` is on, so an
+  # unguarded non-zero exit would kill the script before anything could read the
+  # code. The estimator prints its own detailed diagnosis to stderr on every path,
+  # so the job here is to stop MISLABELING it and name the right remedy -- not to
+  # restate what it already said.
+  rc=0
+  COVER_OUT="$COVER_OUT" bash "$HELPER" || rc=$?
+  case "$rc" in
+    0) ;;
+    1)
+      fail "patch coverage below codecov.yml threshold -- add tests for the uncovered lines.\n      (this gate is a conservative lower bound; Codecov typically reads a few points higher)"
+      ;;
+    2)
+      # Setup/config fault, NOT a coverage verdict: a missing profile, an
+      # unreadable codecov.yml, a malformed threshold. Nothing was measured.
+      fail "patch-coverage could not run (exit 2: setup/config). Read its stderr above -- no coverage figure was produced, so this is not a threshold failure."
+      ;;
+    3)
+      # REFUSED: the validity precondition failed, so the diff scope (committed
+      # HEAD) and the profile (working tree) would describe different versions of
+      # the same file. Deliberately NOT folded into 2 -- 2 legitimately routes to
+      # "this repo has no coverage tooling", and a refusal swallowed there is the
+      # silent skip the estimator's guard exists to end.
+      #
+      # TWO causes with DIFFERENT remedies; the estimator's stderr says which:
+      #   - uncommitted Go changes -> commit them, then re-run
+      #   - `git status` unreadable -> a repo-access fault; the tree state is
+      #     UNDETERMINED, committing fixes nothing, and PATCH_COVERAGE_ALLOW_DIRTY
+      #     does not apply (that branch exits before the override is consulted)
+      fail "patch-coverage REFUSED to measure (exit 3) -- no figure was produced, so this is NOT a threshold failure.\n      Read its stderr above for which precondition failed: commit your changes, or repair git access."
+      ;;
+    *)
+      fail "patch-coverage exited $rc, which this gate does not recognize. Read its stderr above; do not assume a coverage verdict."
+      ;;
+  esac
 else
   echo "    estimator not found at $HELPER"
   echo "    skipping local patch-coverage; Codecov enforces it in CI."

